mp/src/game/client/momentum/ui/HUD/hud_speedometer_data.cpp

```diff
@@ -110,33 +110,54 @@ void SpeedometerData::LoadGamemodeData(GameMode_t gametype)
 
     m_CurrentlyLoadedGamemodeSettings = gametype;
 
-    // get ordering
-    KeyValues *pOrderKV = pGamemodeKV->FindKey("order");
-    if (pOrderKV)
+    // get autolayout. if no custom layout specified, set autolayout on
+    KeyValues *pLayoutKV = pGamemodeKV->FindKey("layout");
+    bool bAutoLayout = pGamemodeKV->GetBool("autolayout", true) || !pLayoutKV;
+    g_pSpeedometer->SetAutoLayout(bAutoLayout);
+    
+    if (!bAutoLayout)
     {
-        auto pOrderList = g_pSpeedometer->GetLabelOrderListPtr();
-        pOrderList->RemoveAll();
-        char szIndex[BUFSIZELOCL];
-        for (unsigned int i = 1; i <= SPEEDOMETER_MAX_LABELS; i++)
+        KeyValuesAD layoutKVTmp(pLayoutKV->MakeCopy());
+        // get control settings then merge them into the custom layout (keeping its settings)
+        KeyValues *pSpeedoSettings = new KeyValues("SpeedoSettings");
```

Comment:
Use KeyValuesAD for any scope-specific keyvalues. (Created and deleted in a single scope)